### PR TITLE
Add virtual destructor

### PR DESCRIPTION
**Pull request by d7d1cd**
C++

### CODE REVIEW

File: include/grpc/event_engine/extensible.h [hunk 1]
```diff
@@ -60,6 +60,7 @@ class Extensible {
   /// if (endpoint != nullptr) endpoint->Process();
   ///
   virtual void* QueryExtension(absl::string_view /*id*/) { return nullptr; }
+  virtual ~Extensible() = default;
```
Comment (yashykt) [inline include/grpc/event_engine/extensible.h:63]: can you move this just after line 26?

**Review by yashykt — APPROVED** [review]

**Comment by ctiller** [comment]
Let's not do this please, and instead add a:
```
protected:
  ~Extensible() = default;
```

**Comment by drfloob** [comment]
+1 to Craig's comment, I just commented a reference on the related issue that explains why: https://github.com/grpc/grpc/issues/37120#issuecomment-2204312257

File: include/grpc/event_engine/extensible.h (hunk 1 above)
Comment (d7d1cd) [reply include/grpc/event_engine/extensible.h]: I did as ctiller suggested

**Review by drfloob — APPROVED** [review]

**Comment by drfloob** [comment]
@d7d1cd Please sign the CLA, and we can move forward.

**Comment by d7d1cd** [comment]
What needs to be done to get the changes into the branch?

**Comment by drfloob** [comment]
> What needs to be done to get the changes into the branch?

Now that you've signed the CLA, I've started the CI jobs for this PR. If all looks good, this should be merged today.

**Comment by drfloob** [comment]
Sanity checks have failed due to clang-format problems. See https://btx.cloud.google.com/invocations/122b491b-8891-4604-bf27-4bf4e5fa47c5/targets. I don't seem to have permission to push a change to your branch, so you'll need to get this sorted. To run clang format, use the script `./tools/distrib/clang_format_code.sh`.

**Comment by d7d1cd** [comment]
@drfloob, I fixed it

**Comment by drfloob** [comment]
I just noticed that this PR targets the 1.65.x branch. We only develop against `master`, the release branches are used for releases & important backports only. I don't seem to have the ability to change the target branch either.

**Comment by d7d1cd** [comment]
I was able to change the target branch to master. However, since I didn’t start with it, now in addition to my change there are a lot of others.

Perhaps I'll close this PR and make the same change in another.

**Pull request closed without merging**